eta/core/data.py

```diff
@@ -1029,3 +1031,11 @@ def optional(cls):
     @classmethod
     def required(cls):
         return ["video_path", "label"]
+
+
+class FeatureArray(NpzWriteable):
```

Comment:
Second comment for discussion.
Do we want to be so tied to Npz?  While sufficient for now, I think a broader approach may be preferred here; perhaps where this inherits a contract and the writer/format is "connected" in some way.

Reply:
Are there any classes currently used to store images as numpy arrays, for example, that I am missing and could be generalized to include this use case?

Reply:
There are not any existing classes that encapsulate binary-type data in "official" formats. We store video and images directly as `.mp4`, `.png`, etc.

Jason's is suggesting that we may consider abstracting this so modules could work with labeled features stored in non-npz formats with the same interface. This is interesting but I would personally wait for the need before acting on it.